packages/http-server/src/http-server.ts

```diff
@@ -4,21 +4,42 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import {createServer, Server, ServerRequest, ServerResponse} from 'http';
+import {createServer as createHttpsServer, Server as HttpsServer} from 'https';
 import {AddressInfo} from 'net';
+import * as util from 'util';
+import * as fs from 'fs';
 import * as pEvent from 'p-event';
 
-export type HttpRequestListener = (
-  req: ServerRequest,
-  res: ServerResponse,
-) => void;
+const readFile = util.promisify(fs.readFile);
+
+export type RequestListener = (req: ServerRequest, res: ServerResponse) => void;
+
+export type CreateServerOptions = {
```

Comment:
What's the difference between `CreateServerOptions` and `ServerOptions`? Could you please explain?

Many properties are shared by both of those types and also `HttpsOptions`, please find a way how to reduce duplication. 

First of all, rework `export type Foo {...}` to `export interface Foo {...}`. Secondly, you can leverage interface inheritance to reduce duplication, for example `interface ServerOptions extend CreateServerOptions`.